cmd/edgectl/cluster.go

```diff
@@ -65,22 +66,28 @@ func (d *Daemon) Connect(p *supervisor.Process, out *Emitter, rai *RunAsInfo, ka
 
 	tmgr, err := NewTrafficManager(p, d.cluster)
 	if err != nil {
-		out.Printf("Failed to connect to traffic manager: %v\n", err)
+		out.Println()
+		out.Println("Unable to connect to the traffic manager in your cluster.")
+		out.Println("The intercept feature will not be available.")
+		out.Println("Error was:", err)
+		// out.Println("Use <some command> to set up the traffic manager.") // FIXME
 	} else {
 		d.trafficMgr = tmgr
 	}
 	return nil
 }
 
 // Disconnect from the connected cluster
-func (d *Daemon) Disconnect(_ *supervisor.Process, out *Emitter) error {
+func (d *Daemon) Disconnect(p *supervisor.Process, out *Emitter) error {
 	// Sanity checks
 	if d.cluster == nil {
 		out.Println("Not connected")
 		return nil
 	}
 
+	_ = d.ClearIntercepts(p)
 	if d.bridge != nil {
+		d.cluster.SetBridgeCheck(nil) // Stop depending on this bridge
```

Comment:
This feels like a bug fix?

Reply:
Yes, this fixes a race condition. I should have called it out as "interesting."